From 7301c9e13c9c684fd4fe706ac1558854a7819a54 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 1 Oct 2019 13:42:11 +0000
Subject: [PATCH 3/4] config: Clear error after submission URL lookup fails

Otherwise we have a dangling pointer causing a crash when the error
value is reused.
---
 src/gclue-config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gclue-config.c b/src/gclue-config.c
index 00dd52e..76222ff 100644
--- a/src/gclue-config.c
+++ b/src/gclue-config.c
@@ -257,7 +257,7 @@ load_wifi_config (GClueConfig *config)
         if (error != NULL) {
                 g_debug ("Failed to get config \"wifi/submission-url\": %s",
                          error->message);
-                g_error_free (error);
+                g_clear_error (&error);
                 priv->wifi_submit_url = g_strdup (DEFAULT_WIFI_SUBMIT_URL);
         }
 
-- 
2.23.0

